api/v1beta1/index/cluster.go

```diff
@@ -30,8 +30,44 @@ import (
 const (
 	// ClusterClassNameField is used by the Cluster controller to index Clusters by ClusterClass name.
 	ClusterClassNameField = "spec.topology.class"
+
+	// ClusterClassRefPath is used by the Cluster controller to index Clusters by ClusterClass name and namespace.
+	ClusterClassRefPath = "spec.topology.classRef"
+
+	// clusterClassRefFmt is used to correctly format class ref index key.
+	clusterClassRefFmt = "%s/%s"
 )
 
+// ByClusterClassRef adds the cluster class name  index to the
+// managers cache.
+func ByClusterClassRef(ctx context.Context, mgr ctrl.Manager) error {
+	if err := mgr.GetCache().IndexField(ctx, &clusterv1.Cluster{},
+		ClusterClassNameField,
+		ClusterByClusterClassRef,
+	); err != nil {
+		return errors.Wrap(err, "error setting index field")
+	}
+	return nil
+}
+
+// ClusterByClusterClassRef contains the logic to index Clusters by ClusterClass name.
+func ClusterByClusterClassRef(o client.Object) []string {
+	cluster, ok := o.(*clusterv1.Cluster)
+	if !ok {
+		panic(fmt.Sprintf("Expected Cluster but got a %T", o))
+	}
+	if cluster.Spec.Topology != nil {
+		key := cluster.GetClassKey()
+		return []string{fmt.Sprintf(clusterClassRefFmt, key.Namespace, key.Name)}
+	}
+	return nil
+}
+
+// ClusterClassRef returns ClusterClass index key to be used for search.
+func ClusterClassRef(o client.Object) string {
```

Comment:
```suggestion
func ClusterClassRef(cc clusterv1.ClusterClass) string {
```
To make this slightly more typesafe